### ISSUE TITLE
Issue to use python binding

### ISSUE THREAD

**Issue by JPLaude** [KDAB/KDDockWidgets#114]
Hello, 

I'm trying to run python example with python binding for windows, but it fails with error:
Traceback (most recent call last):
  File "E:\work\KDDockWidgets\python\examples\main.py", line 18, in <module>
    from PyKDDockWidgets import KDDockWidgets
  File "E:\pv\Lib\site-packages\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
  File "C:\KDAB\KDDockWidgets-1.1.0\PyKDDockWidgets\__init__.py", line 24, in <module>
    from .KDDockWidgets import KDDockWidgets as _priv
  File "E:\pv\Lib\site-packages\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
**ImportError: DLL load failed while importing KDDockWidgets: Le module spécifié est introuvable.**

NOTE that I provided path to PyKDDockWidgets and all dll that it needs, I checked dll loading with dependencies walker.

I built 1.1 branch with:
Visual Studio 2019 compiler version v142/14.26 (14.28 has issue with clong in shiboken_generator)
python 3.9.1
QT 5.14.2

I tried to build for win32 or x64 and I have same issue, I also tried VS2017 without success.

Do you have an idea what I'm doing wrong or do you have any 'full' BKM that could help me generate correctly the python bindings ?

thanks much for your help !

**Comment by renatofilho** [comment]
are you able to import PySide2 modules?

**Comment by JPLaude** [comment]
Yes I can import PySide2
Note I use PySide2, shiboken2 & shiboken2_generator from http://download.qt.io/official_releases/QtForPython/

**Comment by JPLaude** [comment]
I tried to:
  - build pyside2 by myself (5.12.2)
  - KDDockWidgets using built pyside2 (1.1)

I still have same issue, I used the pyside-setup\tools\debug_windows.py tool to try to narrow down the issue and it returns:
```
python e:\work\pyside-setup\tools\debug_windows.py python -c "import PyKDDockWidgets"
[...]
stderr: Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "e:\work\KDDockWidgets\build\out\PyKDDockWidgets\__init__.py", line 24, in <module>
    from .KDDockWidgets import KDDockWidgets as _priv
  File "E:\pv\Lib\site-packages\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
ImportError: DLL load failed while importing KDDockWidgets: Le module spcifi est introuvable.
```
My conclusion is: it finds correctly the PyKDDockWidgets, but not the KDDockWidget.pyd. folder structure is:
PyKDDockWidgets/__init__.py
PyKddockWidgets/KDDockWidgets.pyd

Any clue why it doesn't find the .pyd ?

As a side note PySide2.QtCore is loaded correctly.

**Comment by JPLaude** [comment]
I think I found the issue, it is a dll search path issue, I had to add kddockwidgets' *.dll file path with:
os.add_dll_directory("E:\\work\\KDDockWidgets\\build\\bin") in __init__.py, PATH environement variable is not enough to find dll.

To fix this I think that on python binding installation, you **need** to copy dll to PyKDDockWidgets package, overwise you need to programaticaly add the dll path with os.add_dll_directory() as I did.

Maybe a note in the README.md about it would be good.

**Comment by mkimberling** [comment]
I'm experiencing this error but this time it looks like something else.  kddockwidgets1.dll appears to have been built with the debug Qt library.

Running "dumpbin /dependents kddockwidgets1.dll" yields this set of dependent dll's:
```
dwmapi.dll, Qt5Widgetsd.dll, Qt5Guid.dll, Qt5Cored.dll, USER32.dll, MSVCP140.dll, VCRUNTIME140.dll, VCRUNTIME140_1.dll
api-ms-win-crt-string-l1-1-0.dll, api-ms-win-crt-runtime-l1-1-0.dll, api-ms-win-crt-math-l1-1-0.dll, api-ms-win-crt-heap-l1-1-0.dll, KERNEL32.dll
```
Why did it link against the Debug Qt libraries?
I configured my build with this command line:
cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DKDDockWidgets_STATIC=False -DKDDockWidgets_PYTHON_BINDINGS=True -DCMAKE_PREFIX_PATH=%QT_ROOT% -DCMAKE_INSTALL_PREFIX=..\kd_dock_widgets KDDockWidgets-1.5

I confirmed this is the cause because I was able to get past the error after copying the missing debug qt dll's into the PyKDDockWidgets folder.
